flink-scala/src/main/java/org/apache/flink/api/scala/typeutils/ScalaOptionSerializerSnapshot.java

```diff
@@ -24,7 +24,17 @@
 
 import scala.Option;
 
-/** A {@link TypeSerializerSnapshot} for the Scala {@link OptionSerializer}. */
+/**
+ * A {@link TypeSerializerSnapshot} for the Scala {@link OptionSerializer}.
+ *
+ * @deprecated All Flink Scala APIs are deprecated and will be removed in a future Flink major
+ *     version. You can still build your application in Scala, but you should move to the Java
+ *     version of either the DataStream and/or Table API.
+ * @see <a
+ *     href="https://cwiki.apache.org/confluence/display/FLINK/FLIP-265+Deprecate+and+remove+Scala+API+support">
+ *     FLIP-265 Deprecate and remove Scala API support</a>
+ */
+@Deprecated
 public final class ScalaOptionSerializerSnapshot<E>
```

Comment:
I'm not sure if this and other `SerializerSnapshot` should be treat as public API.

Reply:
Yeah, I think they all should have been `@Internal` -- my reasoning was that removing a `@deprecated` annotation should probably always safe, even between minor releases.  Would you prefer these to be removed?

Reply:
IMO, `@Deprecated` is only marked for public api. So I prefer these to be removed.